drivers/platform/stm32/stm32_spi.c

```diff
@@ -250,6 +270,19 @@ int32_t stm32_spi_write_and_read(struct no_os_spi_desc *desc,
 	gdesc = sdesc->chip_select->extra;
 	SPIx = sdesc->hspi.Instance;
 
+	// Compute a slave ID based on SPI instance and chip select.
+	// If it did not change since last call to stm32_spi_write_and_read,
+	// no need to reconfigure SPI. Otherwise, reconfigure it.
+	slave_id = ((uint64_t)(uintptr_t)sdesc->hspi.Instance << 32) |
+		   sdesc->chip_select->number;
+	if (slave_id != last_slave_id) {
+		ret = stm32_spi_config(desc);
+		if (ret)
+			return ret;
+	}
+
+	last_slave_id = slave_id;
```

Comment:
this can go inside `if (slave_id != last_slave_id)`

Reply:
yeah true, i will move it inside